src/Database/PG/Query/Listen.hs

```diff
@@ -86,20 +84,9 @@ waitForReadReadiness :: PQ.Connection -> ExceptT PGConnErr IO ()
 waitForReadReadiness conn = do
   -- Get file descriptor of underlying socket of a connection
   mFd <- lift $ PQ.socket conn
-  onJust mFd withFd
-  where
-    -- Perform select(2) operation on file descriptor
-    -- to check whether it is ready for reading
-    withFd fd = do
-      selRes <- lift $ do
-        -- Make file descriptors set
-        r <- PS.fromList [fd] --read
-        w <- PS.empty -- write
-        e <- PS.empty -- exception
-        PS.select r w e PS.Never
-      case selRes of
-        PS.Error -> throwError $ PGConnErr "select() failed"
-        _        -> return ()
+  fd <- maybe (throwError $ PGConnErr "connection is not currently open") pure mFd
+  -- Wait for the socket to be ready for reading
+  lift $ threadWaitRead fd
```

Comment:
Capture the `IOError` and wrap it in `PGConnErr`?

Reply:
Will do—is the right thing to do here just `show` the `IOError` and stuff the result inside a `PGConnErr`?

Reply:
Yup. That should be good enough.